### DISCUSSION TITLE
WSUS availability?

### DISCUSSION THREAD

**Discussion by zeekoe** [microsoft/openjdk#144] [Q&A] [answered]
In our company, we distribute updates with WSUS. OpenJDK is not available there, and is not in the Microsoft Update catalog: https://www.catalog.update.microsoft.com
Is there a recommended way to get Microsoft OpenJDK available in WSUS? Or could we use winget on servers running newer server OSes, even though they don't have internet access?

**Comment by karianna** [comment]
@zeekoe How do you currently deploy/update Java?

**Reply by zeekoe** [reply]
On some servers, manually. But that's a roadblock in installing additional tooling ("very nice to have with workaround" and different value for different stakeholders) on a new server, which could be taken away with automated upgrades.

**Comment by enRchi** [comment]
Would also make sense to me that Microsoft would make it possible to update this with WSUS. This would make Microsofts Version of the OpenJDK the easiest to update under Windows. Or alternatively provide it for download in the MS Store. (But afaik then it would not be available for Windows Server)

**Comment by karianna** [comment, marked as answer]
At this stage there are no plans to provide updates via this mechanism

**Reply by jkohutOMF** [reply]
I hope MS reconsiders this. Many companies see how much eaiser OpenSource is to update on Linux and are moving in that direction. Java, Apache HTTPD, Apache Tomcat, and NodeJS are a few things that would help Windows get support application updates. If nothing else a "hook" into the WSUS update process would be extremly helpful.

**Comment by tsax6010** [comment]
This is an opportunity for Microsoft to help bridge opensource to Windows by utilizing their update mechanism to keep historically vulnerable code up to date. What is the point of Microsoft diving into the JDK open project if they are just going to perpetuate the cesspool of Java vulnerabilities made possible by horrible manually patching practices? Comon, WSUS, Arc, something needs to be integrated here with your builds of OpenJDK to allow a more seamless and secure integration with your own operating systems.
